src/test/regress/expected/coordinator_tables.out

```diff
@@ -0,0 +1,230 @@
+ create_coordinator_table
+---------------------------------------------------------------------
+
+(1 row)
+
+CREATE TABLE coordinator_table_2 (a int primary key);
+SELECT create_coordinator_table('coordinator_table_2');
+NOTICE:  executing the command locally: SELECT worker_apply_shard_ddl_command (1504024, 'coordinator_tables_test_schema', 'CREATE TABLE coordinator_tables_test_schema.coordinator_table_2 (a integer NOT NULL)');SELECT worker_apply_shard_ddl_command (1504024, 'coordinator_tables_test_schema', 'ALTER TABLE coordinator_tables_test_schema.coordinator_table_2 OWNER TO postgres');SELECT worker_apply_shard_ddl_command (1504024, 'coordinator_tables_test_schema', 'ALTER TABLE coordinator_tables_test_schema.coordinator_table_2 ADD CONSTRAINT coordinator_table_2_pkey PRIMARY KEY (a)')
+ create_coordinator_table
+---------------------------------------------------------------------
+
+(1 row)
+
+CREATE TABLE local_table (a int primary key);
+CREATE TABLE distributed_table (a int primary key);
+SELECT create_distributed_table('distributed_table', 'a');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+CREATE TABLE reference_table (a int primary key);
+SELECT create_reference_table('reference_table');
+ create_reference_table
+---------------------------------------------------------------------
+
+(1 row)
+
+-- any foreign key between coordinator tables and other tables cannot be set for now
+-- each should error out (for now meaningless error messages)
+-- between coordinator tables
+ALTER TABLE coordinator_table_1 ADD CONSTRAINT fkey_c_to_c FOREIGN KEY(a) references coordinator_table_2(a);
+ERROR:  cannot create foreign key constraint since relations are not colocated or not referencing a reference table
+-- between coordinator tables and reference tables
+ALTER TABLE coordinator_table_1 ADD CONSTRAINT fkey_c_to_ref FOREIGN KEY(a) references reference_table(a);
+ERROR:  cannot create foreign key constraint since relations are not colocated or not referencing a reference table
+ALTER TABLE reference_table ADD CONSTRAINT fkey_ref_to_c FOREIGN KEY(a) references coordinator_table_1(a);
+ERROR:  cannot create foreign key constraint since foreign keys from reference tables to distributed tables are not supported
+-- between coordinator tables and distributed tables
+ALTER TABLE coordinator_table_1 ADD CONSTRAINT fkey_c_to_dist FOREIGN KEY(a) references distributed_table(a);
+ERROR:  cannot create foreign key constraint since relations are not colocated or not referencing a reference table
+ALTER TABLE distributed_table ADD CONSTRAINT fkey_dist_to_c FOREIGN KEY(a) references coordinator_table_1(a);
+ERROR:  cannot create foreign key constraint since relations are not colocated or not referencing a reference table
+-- between coordinator tables and local tables
+ALTER TABLE coordinator_table_1 ADD CONSTRAINT fkey_c_to_local FOREIGN KEY(a) references local_table(a);
```

Comment:
One more set of tests I'd like to have is show that `mark_tables_colocated`  fails as that might be very confusing otherwise. It all works expected, just let's add so that we don't break that anytime future:

SELECT mark_tables_colocated('local_1', ARRAY['local_2']);
SELECT mark_tables_colocated('local_1', ARRAY['reference']);
SELECT mark_tables_colocated('local_1', ARRAY['distributed']);
 SELECT mark_tables_colocated('distributed', ARRAY['local']);
SELECT mark_tables_colocated('reference', ARRAY['local']);